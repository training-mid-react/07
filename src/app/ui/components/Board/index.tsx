import { FC } from 'react';
import './styles.scss';
import { Square } from '../Square';
import { ROWS } from '@core/constants';
import { OptionsColors } from '@core/types';

interface Props {
    board: OptionsColors[][];
    handleClick: (col: number) => void;
    fallingCells: string[];
}
export const Board: FC<Props> = ({ board, handleClick, fallingCells }) => (
    <div className="board">
        {board?.map((_, col) => (
            <div key={col} className="board__column">
                {Array.from({ length: ROWS }, (_, row) => (
                    <Square
                        key={`${row}-${col}`}
                        onClick={() => handleClick(col)}
                        value={board[row][col]}
                        isFalling={fallingCells.includes(`${row}-${col}`)}
                    />
                ))}
            </div>
        ))}
    </div>
);
