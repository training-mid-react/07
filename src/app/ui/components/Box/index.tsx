import { CELL } from "@/app/core/interfaces/game.interface";
import "./style.scss"

interface Props {
    value: CELL;
    onClick: () => void;
}

const Box = ({ value, onClick }: Props) => {
    return (
        <div data-testid="box" className="box" onClick={onClick}>
            {
                value && <div style={{backgroundColor: value.color}} className="box__chip"></div>
            }
        </div>
    );
}

export default Box;