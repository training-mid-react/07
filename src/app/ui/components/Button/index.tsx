import React from 'react';
import './style.scss';

interface ButtonProps {
  children: React.ReactNode;
  className?: string;
  onClick: () => void;
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({ children, onClick, disabled, className }) => {
  return (
    <button data-testid='button-component' className={className || 'custom-button'} onClick={onClick} disabled={disabled}>
      {children}
    </button>
  );
};

export default Button;
