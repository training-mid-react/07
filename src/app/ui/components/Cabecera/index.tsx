
import "./style.scss";

interface Props {
    currentPlayer: number
}

const Cabecera = ({ currentPlayer }: Props) => {
    return (
        <>
            <section className="cabecera">
                <div>
                    <h2>Jugador 1</h2>
                    <div className="ficha-cabecera ficha-cabecera--jugador1"></div>
                </div>
                <div>
                    <h2>Jugador 2</h2>
                    <div className="ficha-cabecera ficha-cabecera--jugador2"></div>
                </div>
            </section>
            <h3>Turno del jugador: {currentPlayer}</h3>
        </>

    )
}

export default Cabecera 