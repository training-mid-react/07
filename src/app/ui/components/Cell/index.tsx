import { FC } from 'react';
import { PLAYER } from '../../../../core/interfaces';
import './style.scss';

export interface CellProps {
  value: PLAYER | null;
}

export const Cell: FC<CellProps> = ({ value }) => {

  return (
    <>
    <div className={`cell cell__${value}`} />
    <div className='cell__frame'></div>
    </>
  );
};
