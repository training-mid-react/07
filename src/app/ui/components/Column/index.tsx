import './style.scss';
import { useDropCell } from '../../../core/hooks/useDropCell';

interface ColumnProps {
  columnIndex: number;
  dropPiece: (column: number) => void;
  columnData: number[];
}

function Column({ columnIndex, dropPiece, columnData }: ColumnProps) {
  const { handleColumnClick, droppingCell } = useDropCell({ columnIndex, dropPiece, columnData })
  return (
    <div className="column" onClick={handleColumnClick}>
      {columnData.map((cell, rowIndex) => {
        const isDropping = droppingCell === rowIndex;
        const cellClass = `cell ${cell === 1 ? 'cell--player1' : cell === 2 ? 'cell--player2' : ''} ${isDropping ? 'cell--dropping' : ''}`;
        return (
          <div
            key={rowIndex}
            className={cellClass}
          />
        );
      })}
    </div>
  );
};

export default Column;
