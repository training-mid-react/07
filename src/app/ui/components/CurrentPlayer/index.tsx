import './style.scss';

interface ICurrentPlayerProps {
  currentPlayer: number;
}

export const CurrentPlayer = ({ currentPlayer }: ICurrentPlayerProps) => {
  return (
    <div className="current-player">
      Es el turno del Jugador: 
      <span className={currentPlayer === 1 ? 'current-player__player1' : 'current-player__player2'}>
        {currentPlayer}
      </span>
    </div>
  );
};
