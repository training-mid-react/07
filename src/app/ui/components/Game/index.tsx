import { HeaderBoard } from '../HeaderBoard';
import { Board } from '../Board';
import { FC } from 'react';
import { IState } from '@core/interfaces';

export const Game: FC<{
    state: IState;
    onClickResetButton: () => void;
    handleClick: (col: number) => void;
    fallingCells: string[];
}> = ({ state, onClickResetButton, handleClick, fallingCells }) => (
    <>
        <HeaderBoard state={state} onClickResetButton={onClickResetButton} />
        <Board
            handleClick={handleClick}
            fallingCells={fallingCells}
            board={state.board}
        />
    </>
);
