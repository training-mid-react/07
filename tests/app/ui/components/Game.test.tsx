import { render, screen } from '@testing-library/react';
import { Game } from '../../../../src/app/ui/components/Game';
import { describe, expect, test, vi } from 'vitest';
import React from 'react';

describe('Game', () => {
    const mockOnClickResetButton = vi.fn();
    const mockOnHandleClick = vi.fn();
    const mockState = {
        state: {
            players: [
                {
                    id: 1,
                    name: 'Jugador 1',
                    color: 'red',
                    isCurrentPlayer: true,
                },
                {
                    id: 2,
                    name: 'Jugador 2',
                    color: 'blue',
                    isCurrentPlayer: false,
                },
            ],
            winner: '',
            board: [[]],
        },
        dispatch: vi.fn(),
    };

    test('Match snapshoot', () => {
        const { asFragment } = render(
            <Game
                state={mockState}
                onClickResetButton={mockOnClickResetButton}
                handleClick={mockOnHandleClick}
                fallingCells={[]}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
