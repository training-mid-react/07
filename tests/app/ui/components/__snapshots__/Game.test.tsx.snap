// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game > Match snapshoot 1`] = `
<DocumentFragment>
  <div
    class="board-layout__header"
  >
    <section>
      <h3>
        Jugadores:
      </h3>
      <ul />
    </section>
    <h1>
      undefined es tu turno
    </h1>
    <button>
      Reiniciar juego
    </button>
  </div>
  <div
    class="board"
  />
</DocumentFragment>
`;
